src/LibraryUtilities.ts

```diff
@@ -510,13 +510,21 @@ export function buildLibrarySectionsFromLayoutSpecs(loadedTypes: any, layoutSpec
 
 // Remove empty non-leaf nodes from items
 export function removeEmptyNodes(items: ItemData[]) {
+    let itemRemoved = false;
+
     items.forEach((item, index, items) => {
         if (item.childItems.length > 0) {
-            removeEmptyNodes(item.childItems);
+            if (removeEmptyNodes(item.childItems)) {
+                removeEmptyNodes(items);
```

Comment:
Unnecessary computation is happening here. Assuming there are 1000 items, the last one has some child items removed, then all the first 999 items will be tried again.